test/test-ghe-backup.sh

```diff
@@ -344,3 +345,74 @@ begin_test "ghe-backup missing directories or files on source appliance"
     verify_all_backedup_data
 )
 end_test
+
+if [ "$(uname)" == "Darwin" ]; then
+    timeout() {
+        ruby -rtimeout -e 'duration = ARGV.shift.to_i; Timeout::timeout(duration) { system(*ARGV) }' "$@"
+    }
+fi
```

Comment:
🤔 this is pulling in a pretty large (time, effort and bytes of code) dependency on Ruby (which won't ship with macOS in future) just to run tests on macOS hosts which may not have `coreutils` installed. We're already assuming users have `coreutils` installed because of our Linux assumptions. How about making `coreutils` a macOS requirement too and remove this function?